instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/TracingRabbitListenerAdviceTest.java

```diff
@@ -196,6 +203,72 @@
       .containsExactly(error);
   }
 
+  @Test public void batch_starts_new_trace_if_none_exists() throws Throwable {
+    // non traced that listener is a new trace
+    onBatchMessageConsumed(Arrays.asList(MessageBuilder.withBody(new byte[0]).build(),
+      MessageBuilder.withBody(new byte[0]).build()));
+
+    assertThat(spans)
+      .extracting(MutableSpan::kind)
+      .containsExactly(CONSUMER, null);
+  }
+
+  @Test public void batch_continue_parent_trace() throws Throwable {
+    MessageProperties props = new MessageProperties();
+    props.setHeader("X-B3-TraceId", TRACE_ID);
+    props.setHeader("X-B3-SpanId", SPAN_ID);
+    props.setHeader("X-B3-ParentSpanId", PARENT_ID);
+    props.setHeader("X-B3-Sampled", SAMPLED);
+
+    Message message = MessageBuilder.withBody(new byte[0]).andProperties(props).build();
+    MessageProperties props2 = new MessageProperties();
+    props2.setHeader("X-B3-TraceId", TRACE_ID_2);
+    props2.setHeader("X-B3-SpanId", SPAN_ID_2);
+    props2.setHeader("X-B3-ParentSpanId", PARENT_ID_2);
+    props2.setHeader("X-B3-Sampled", SAMPLED);
+    Message message2 = MessageBuilder.withBody(new byte[0]).andProperties(props2).build();
+    onBatchMessageConsumed(Arrays.asList(message, message2));
+
+    // two traced that listener continues first trace but TODO we aren't handling the second.
+    assertThat(spans.get(0))
```

Comment:
last thing from me on this.. I see you are only checking spans.get(0) the reported trace.

though I don't see any problems.. maybe ensure the B3 headers are cleared in all messages in each test after onBatchMessageConsumed similar to other tests

```
    // cleared the headers to later work doesn't try to use the old parent
    assertThat(message.getMessageProperties().getHeaders()).isEmpty();
```

Reply:
Thanks. fixed at last commit.